pkg/oauth/user.go

```diff
@@ -207,16 +211,53 @@ func (s *server) Login(c echo.Context) error {
 	return c.NoContent(http.StatusNoContent)
 }
 
+var errInvalidLogoutRedirectURI = errors.DefinePermissionDenied("invalid_logout_redirect_uri", "the redirect URI did not match the one(s) defined in the client")
+
 func (s *server) Logout(c echo.Context) error {
 	ctx := c.Request().Context()
-	session, err := s.getSession(c)
+	accessToken := c.QueryParam("access_token")
+	_, id, _, err := auth.SplitToken(accessToken)
+	if err != nil {
+		return err
+	}
+	at, err := s.store.GetAccessToken(ctx, id)
 	if err != nil {
 		return err
 	}
-	events.Publish(evtUserLogout(ctx, session.UserIdentifiers, nil))
-	if err = s.store.DeleteSession(ctx, &session.UserIdentifiers, session.SessionID); err != nil {
+	sessions, err := s.store.FindSessions(ctx, &at.UserIDs)
+	if err != nil {
+		return err
+	}
+	if err = s.store.DeleteAccessToken(ctx, id); err != nil {
 		return err
 	}
 	s.removeAuthCookie(c)
-	return c.NoContent(http.StatusNoContent)
+	events.Publish(evtUserLogout(ctx, at.UserIDs, nil))
+	for _, session := range sessions {
+		if err = s.store.DeleteSession(ctx, &at.UserIDs, session.SessionID); err != nil {
```

Comment:
I don't think this is what a user wants. If I logout using my phone, I don't want to also be logged out from my laptop.

Reply:
Right, I went with the easy way for now. To be able to do this, we'd need to know the session that was used when obtaining the access token. To do this, we could either:
- pass the session id to the console together with the access token
- store the session id in the access token store when creating it